0099-group-challenge.md

```diff
@@ -0,0 +1,148 @@
+Alternative suggestions welcomed.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- Pioneer spot rewards: we need to figure out the incentive structure for people to add gateways and light gateways in locations that can't see any other gateways.
+- Light gateway information services: we need to figure out the best way to make the necessary information for a PoC challenge available to light gateways.
+- Voting Transactions: Are they needed, or can we get away with simply slashing group members that aren't carrying their weight in terms of making challenges happen.
+- Greedy Challengers: A node could challenge everyone in an attempt to "claim" all of the PoC challenges for this round either for censorship purposes or to force slashing for other nodes (so as to stay longer in the consensus group).  The request/reporting mechanism needs careful design to prevent this as well.
+
+# Deployment Impact
+[deployment-impact]: #deployment-impact
+
+**Describe how this design will be deployed and any potential imapact it may have on current users of this project.**
+
+There should be fairly little impact on the current network at this stage of the process.
+We'll activate a chain var and behavior will change.
+However, network incentives will change, perhaps a lot.
+Since there will no longer be any challenges, lone-wolf nodes will no longer have the same incentive to participate or pioneer new regions.
+
+Additionally, many users (and developers) use challenge recency as a heartbeat metric.
+We will need to develop alternative metrics to for assessing the health of the network at large.
+
+A more positive impact of this changeover should be that longer PoC paths outside of cheat clusters should be more feasible, as they will require less from the participating nodes.
+
+# Success Metrics
+[success-metrics]: #success-metrics
+
+**What metrics can be used to measure the success of this design?**
+
+Primarily: PoC rate success. The PoC rate shouldn't fall when we enable this design.
+
+Secondarily: Do people actually add light gateways to the network?  How many?  Was this feature really needed for this adoption?
+
+# Appendix A: libp2p Services
+[appendix-a]: #appendix-a
+
+Currently the network is a sea of undifferentiated nodes, all running more or less the same version of erlang lip2p.
+Under this proposal, we will be moving to a network that has nodes with differing capacities (to some extent, we have this today with seed nodes).
+Given that many of these nodes will have small memory and little to no persistent storage, it's probable that we'll need to be able to make informed decisions about what network peers we retain knowledge about.
+In order to inform those decisions, I think that libp2p peers should start to declare some sort of service information, e.g. "can answer ARP requests", "full node", "poc capable", etc.
```

Comment:
libp2p already allows nodes to advertise the "handler protocols" available on a connection as part of the stream negotiation (there's a `ls` command)

Reply:
Yeah, I meant to have it serialized in the peer, so you could look it up/index it locally without having to contact the remote node.